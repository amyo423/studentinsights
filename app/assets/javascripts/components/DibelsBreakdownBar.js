import React from 'react';

// Visual component showing a horizontal bar broken down into three colors
// showing percent of students at different DIBELS levels.
export default class DibelsBreakdownBar extends React.Component {
  totalCount() {
    const {coreCount, strategicCount, intensiveCount} = this.props;
    return coreCount + strategicCount + intensiveCount;
  }

  scale(count) {
    const totalCount = this.totalCount();
    return `${Math.ceil(100 * count / totalCount)}%`;
  }

  render() {
    const {height, coreCount, strategicCount, intensiveCount, style = {}} = this.props;    
    return (
      <div className="DibelsBreakdownBar" style={{height, ...style}}>
        <div style={{position: 'relative', width: '100%', height}}>
          {this.renderBarAndLabel({
            left: 0,
            width: coreCount,
            color: 'green'
          })}
          {this.renderBarAndLabel({
            left: coreCount,
            width: strategicCount,
            color: 'orange'
          })}
          {this.renderBarAndLabel({
            left: coreCount + strategicCount,
            width: intensiveCount,
            color: 'red'
          })}
        </div>
      </div>
    );
  }


  renderBarAndLabel({left, width, color}) {
    const {height, labelTop} = this.props;
    const opacity = 0.5;
    const fontSize = 10;

    if (width === 0) return;
    return (
      <div>
        <div style={{
          opacity,
          fontSize,
          position: 'absolute',
          background: color,
          left: this.scale(left),
          width: this.scale(width),
          height
        }}>{'\u00A0'}</div>
        {width > 0 && 
          <div style={{
            opacity,
            fontSize,
            color,
            position: 'absolute',
            textAlign: 'right',
            left: this.scale(left),
            width: this.scale(width),
            top: labelTop,
            paddingRight: 1
          }}>{width}</div>}
      </div>
    );
  }
}

DibelsBreakdownBar.propTypes = {
  coreCount: React.PropTypes.number.isRequired,
  strategicCount: React.PropTypes.number.isRequired,
  intensiveCount: React.PropTypes.number.isRequired,
  height: React.PropTypes.number.isRequired,
  labelTop: React.PropTypes.number.isRequired,
  style: React.PropTypes.object
};
