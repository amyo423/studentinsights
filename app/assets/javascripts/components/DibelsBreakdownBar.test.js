import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import DibelsBreakdownBar from './DibelsBreakdownBar';


export function testProps(props) {
  return {
    coreCount: 7,
    strategicCount: 3,
    intensiveCount: 2,
    height: 50,
    labelTop: 55,
    ...props
  };
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  const props = testProps();
  ReactDOM.render(<DibelsBreakdownBar {...props} />, el);
});

it('snapshots', () => {
  const props = testProps();
  const tree = renderer
    .create(<DibelsBreakdownBar {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});