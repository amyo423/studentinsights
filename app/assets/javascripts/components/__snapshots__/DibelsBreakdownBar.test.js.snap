// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  className="DibelsBreakdownBar"
  style={
    Object {
      "height": 50,
    }
  }
>
  <div
    style={
      Object {
        "height": 50,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div>
      <div
        style={
          Object {
            "background": "green",
            "fontSize": 10,
            "height": 50,
            "left": "0%",
            "opacity": 0.5,
            "position": "absolute",
            "width": "59%",
          }
        }
      >
         
      </div>
      <div
        style={
          Object {
            "color": "green",
            "fontSize": 10,
            "left": "0%",
            "opacity": 0.5,
            "paddingRight": 1,
            "position": "absolute",
            "textAlign": "right",
            "top": 55,
            "width": "59%",
          }
        }
      >
        7
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "background": "orange",
            "fontSize": 10,
            "height": 50,
            "left": "59%",
            "opacity": 0.5,
            "position": "absolute",
            "width": "25%",
          }
        }
      >
         
      </div>
      <div
        style={
          Object {
            "color": "orange",
            "fontSize": 10,
            "left": "59%",
            "opacity": 0.5,
            "paddingRight": 1,
            "position": "absolute",
            "textAlign": "right",
            "top": 55,
            "width": "25%",
          }
        }
      >
        3
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "background": "red",
            "fontSize": 10,
            "height": 50,
            "left": "84%",
            "opacity": 0.5,
            "position": "absolute",
            "width": "17%",
          }
        }
      >
         
      </div>
      <div
        style={
          Object {
            "color": "red",
            "fontSize": 10,
            "left": "84%",
            "opacity": 0.5,
            "paddingRight": 1,
            "position": "absolute",
            "textAlign": "right",
            "top": 55,
            "width": "17%",
          }
        }
      >
        2
      </div>
    </div>
  </div>
</div>
`;
